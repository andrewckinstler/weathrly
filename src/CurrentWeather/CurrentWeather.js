import React from 'react';
import './CurrentWeather.scss';
import moment from 'moment';

export const CurrentWeather = props => {
  const { name, weather, main, dt } = props
  const date = new Date(dt * 1000);
  const icon = weather[0].icon

  return (
    <section>
      <div className='today'>
        <div className='current-container'>
          <h2>{name}</h2>
          <p className='date'>{moment(date).calendar()}</p>
          <p className='current-temp'>{main.temp.toFixed(0)}°</p>
          <p>{weather[0].main}</p>
        </div>
        <div className='minmax'>
          <img src={`http://openweathermap.org/img/wn/${icon}@2x.png`} alt='current condition' />
          <p><span className='bold'>H:</span> {main.temp_max.toFixed(0)}° / <span className='bold'>L:</span> {main.temp_min.toFixed(0)}°</p>
          <p>{weather[0].description}</p>
        </div>
      </div>
    </section>
  )
};