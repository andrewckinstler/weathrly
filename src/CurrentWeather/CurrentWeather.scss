section {
  border-radius: 10px;
  margin: 75px 100px 50px 100px;
  padding: 20px 75px;
  background-color: #fff;

  .bold {
    font-weight: bold;
  }

  .minmax {
    margin-top: -20px;
    text-align: center;
  }

  .today{
    display: flex;
    justify-content: space-evenly;


    .current-container {
      text-align: center;

      h2 {
        font-size: 2.5em;
      }

      .date {
        margin-bottom: 5px;
      }

      .current-temp {
        display: block;
        font-size: 2em;
      }
    }
  } 
}