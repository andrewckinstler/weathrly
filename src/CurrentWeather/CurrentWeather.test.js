import { shallow } from 'enzyme';
import React from 'react';
import { CurrentWeather } from './CurrentWeather';

describe('Current weather card', () => {
  it('should match a snapshot', () => {
    const mockData = {
      name: 'Colorado',
      weather: [
        {
            "id": 521,
            "main": "Rain",
            "description": "shower rain",
            "icon": "09d"
        },
        {
            "id": 621,
            "main": "Snow",
            "description": "shower snow",
            "icon": "13d"
        }
    ],
      main: {
        "temp": 31.68,
        "feels_like": 24.69,
        "temp_min": 26.01,
        "temp_max": 41,
        "pressure": 1016,
        "humidity": 59
    },
      dt: 1580330086
    }
    const wrapper = shallow(<CurrentWeather {...mockData}/>)

    expect(wrapper).toMatchSnapshot()
  })
})