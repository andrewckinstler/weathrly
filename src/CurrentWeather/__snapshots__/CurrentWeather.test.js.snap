// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current weather card should match a snapshot 1`] = `
<section>
  <div
    className="today"
  >
    <div
      className="current-container"
    >
      <h2>
        Colorado
      </h2>
      <p
        className="date"
      >
        Yesterday at 1:34 PM
      </p>
      <p
        className="current-temp"
      >
        32
        °
      </p>
      <p>
        Rain
      </p>
    </div>
    <div
      className="minmax"
    >
      <img
        alt="current condition"
        src="http://openweathermap.org/img/wn/09d@2x.png"
      />
      <p>
        <span
          className="bold"
        >
          H:
        </span>
         
        41
        ° / 
        <span
          className="bold"
        >
          L:
        </span>
         
        26
        °
      </p>
      <p>
        shower rain
      </p>
    </div>
  </div>
</section>
`;
